lib/topics/state_and_behaviour.rb

```diff
@@ -2,4 +2,27 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # class Car
+  class Car
+    attr_reader :color, :year, :model, :current_speed
+
+    def initialize(car_attributes)
+      @color = car_attributes.fetch(:color, 'black')
+      @year  = car_attributes.fetch(:year, 2017)
+      @model = car_attributes.fetch(:model, 'tesla_model_3')
+      @current_speed = nil
+    end
+
+    def self.default_car
+      Car.new
+    end
+
+    def speed_up(speed)
+      @current_speed = speed
```

Comment:
maybe increase speed? `@current_speed += speed` (and try `more_speed` as argument name. more human readable)